lock-impl/src/main/java/com/palantir/lock/logger/LockStateYamlWriter.java

```diff
@@ -84,4 +87,11 @@ private static DumperOptions getDumperOptions() {
     public void close() throws IOException {
         writer.close();
     }
+
+    private static class LockDescriptorAwareRepresenter extends Representer {
+        LockDescriptorAwareRepresenter() {
+            super();
+            this.representers.put(LockDescriptor.class, data -> representScalar(Tag.STR, data.toString()));
```

Comment:
do we want `getLockIdAsString` here?
also, given that we end up converting to a string anyway, is it actually an improvement to use `LockDescriptor` elsewhere?